infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationSourceManager.java

```diff
@@ -165,7 +150,8 @@ public void stopLogReplication() {
      * Termination of the Log Replication State Machine, to enable replication a JVM restart is required.
      */
     public void shutdown() {
-        if(runtime.isShutdown()) {
+        // Shama check and note which aare the 2 places
+        if(isShutdown) {
```

Comment:
Why not to depend on runtime rather than class member variable?

Reply:
since there is 1 runtime shared amongst all FSMs, we cannot shut it down like before. So we rely on another variable to make the shutdown idempotent

Reply:
can we use replicationContext.getCorfuRuntime().isShutdown()?